src/System.Net.WebSockets.Client/tests/CancelTest.cs

```diff
@@ -130,14 +129,9 @@ public class CancelTest : ClientWebSocketTestBase
                 var recvBuffer = new byte[100];
                 var segment = new ArraySegment<byte>(recvBuffer);
 
-                try
-                {
-                    await cws.ReceiveAsync(segment, cts.Token);
-                    Assert.True(false, "Receive should not complete.");
-                }
-                catch (OperationCanceledException) { }
-                catch (ObjectDisposedException) { }
-                catch (WebSocketException) { }
+                Exception e = await Assert.ThrowsAnyAsync<Exception>(() => cws.ReceiveAsync(segment, cts.Token));
+                Assert.True(e is OperationCanceledException || e is ObjectDisposedException || e is WebSocketException,
+                    $"Unexpected exception type {e.GetType()}");
```

Comment:
Hmm, yeah, I can see that.
Actually, after poking around a little more, I think we can just expect `WebSocketException`:
 - `ObjectDisposedException` in this context is probably a holdover from the previous issues with tokens.
 - `OperationCanceledException` is only thrown if the token is canceled before it starts `await`ing (during setup) - it's not thrown during `Receive` otherwise (from the look of it, only during `Send`).

An OCE would thus be a sign that the test didn't get into a necessary state to begin with, and should likely be re-run.